VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -13,21 +13,43 @@
     private DiscussionTextUtils() {
     }
 
-    public static String getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources) {
-        String text = authorData.getAuthor() + " " + DateUtils.getRelativeTimeSpanString(
+    public enum AuthorAttributionType {
+        DISCUSSION,
+        QUESTION,
+        COMMENT
+    }
+
+    public static CharSequence getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources, @NonNull AuthorAttributionType attributionType) {
+        @StringRes int stringRes;
+        switch (attributionType) {
+            case DISCUSSION:
+                stringRes = R.string.discussion_thread_attribution;
+                break;
+            case QUESTION:
+                stringRes = R.string.question_thread_attribution;
+                break;
+            case COMMENT:
+                stringRes = R.string.comment_attribution;
+                break;
+            default:
+                throw new IllegalArgumentException(attributionType.name());
+        }
+
+        final CharSequence formattedTime = DateUtils.getRelativeTimeSpanString(
                 authorData.getCreatedAt().getTime(),
                 System.currentTimeMillis(),
                 DateUtils.MINUTE_IN_MILLIS,
                 DateUtils.FORMAT_ABBREV_RELATIVE);
 
+        String authorLabel = "";
         if (authorData.getAuthorLabel() == PinnedAuthor.STAFF) {
-            text += "  " + resources.getString(R.string.discussion_priviledged_author_label_staff);
+            authorLabel = resources.getString(R.string.discussion_priviledged_author_label_staff);
 
         } else if (authorData.getAuthorLabel() == PinnedAuthor.COMMUNITY_TA) {
-            text = "  " + resources.getString(R.string.discussion_priviledged_author_label_ta);
+            authorLabel = resources.getString(R.string.discussion_priviledged_author_label_ta);
         }
 
-        return text;
+        return trim(resources.getString(stringRes, formattedTime, authorData.getAuthor(), authorLabel.toUpperCase(resources.getConfiguration().locale)));
```

Comment:
>I meant to use CharSequence (now fixed), not a String, which is why we have the custom trim method.

Sorry I don't get it. You're using `resources.getString()`, so you'll be getting a `String` (which is an instance `CharSequence`), on which you can then call `trim()`. I don't think you can do formatting in `CharSequence` anyway.